script/workflowDeploymentAndSetupScript/DeployAndSetupQueueBasedPIM.s.sol

```diff
@@ -0,0 +1,862 @@
+        // Deploy workflow
+
+        // Get the orchestrator factory
+        IOrchestratorFactory_v1 orchestratorFactory =
+            IOrchestratorFactory_v1(getDeployedOrchestratorFactory());
+
+        console2.log("  Deploying Inverter Protocol workflow...");
+        // Create workflow
+        vm.startBroadcast(deployerPrivateKey);
+        _orchestrator = orchestratorFactory.createOrchestrator(
+            _getWorkflowConfig(),
+            _getFundingManagerConfig(),
+            _getAuthorizerConfig(),
+            _getPaymentProcessorConfig(),
+            _getOptionalModulesConfigs()
+        );
+        vm.stopBroadcast();
+
+        // ---------------------------------------------------------------------
+        // Store workflow addresses
+
+        // Store modules
+        // Store funding manager
+        _fundingManager =
+            FM_PC_Oracle_Redeeming_v1(address(_orchestrator.fundingManager()));
+        // Store Authorizer
+        _authorizer = AUT_Roles_v1(address(_orchestrator.authorizer()));
+        // Store Payment Processor
+        _paymentProcessor = PP_Queue_ManualExecution_v1(
+            address(_orchestrator.paymentProcessor())
+        );
+        // Store Oracle module
+        _oracleModule =
+            LM_Oracle_Permissioned_v1(address(_orchestrator.listModules()[0]));
+
+        // ---------------------------------------------------------------------
+        // Validate workflow deployment
+
+        console2.log("  Validating deployed workflow...");
+        _validateWorkflowDeployment();
+        console2.log("  [OK] Workflow deployed successfully\n");
+
+        // ---------------------------------------------------------------------
+        // Log workflow addresses
+        console2.log("Deployed Contract Addresses");
+        console2.log("---------------------------");
+        _logAddress("  * Orchestrator", address(_orchestrator));
+        _logAddress("  * Funding Manager", address(_fundingManager));
+        _logAddress("  * Authorizer", address(_authorizer));
+        _logAddress("  * Payment Processor", address(_paymentProcessor));
+        _logAddress("  * Oracle Module", address(_oracleModule));
+        _logAddress("  * Issuance Token", address(_issuanceToken));
+        console2.log("\n");
+    }
+
+    // -------------------------------------------------------------------------
+    // Setup functions
+
+    function _setupWorkflowAdmin() internal {
+        console2.log("Administrative Setup");
+        console2.log("--------------------");
+
+        console2.log("  Setting up workflow admin...");
+        vm.startBroadcast(deployerPrivateKey);
+        _authorizer.grantGlobalRole(_authorizer.getAdminRole(), _workflowAdmin);
+        require(
+            _authorizer.checkForRole(
+                _authorizer.generateRoleId(
+                    address(_orchestrator), _authorizer.getAdminRole()
```

Comment:
After the change above, you'd also just use `_authorizer.getAdminRole()` here instead of generating.